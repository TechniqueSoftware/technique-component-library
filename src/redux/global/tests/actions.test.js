import { GLOBAL_ACTIONS } from '..'

describe('GLOBAL ACTIONS', () => {
  describe('Loading', () => {
    it('finishes', () => {
      expect(GLOBAL_ACTIONS.LOADING.FINISHED()).toMatchSnapshot()
    })
  })

  describe('User', () => {
    it('sets', () => {
      expect(GLOBAL_ACTIONS.USER.SET_LOGGED_IN({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('gets', () => {
      expect(GLOBAL_ACTIONS.USER.GET({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('succeeds', () => {
      expect(GLOBAL_ACTIONS.USER.SUCCESS({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('errors', () => {
      expect(GLOBAL_ACTIONS.USER.ERROR({
        error: 'some error here'
      })).toMatchSnapshot()
    })

    it('clears', () => {
      expect(GLOBAL_ACTIONS.USER.CLEAR()).toMatchSnapshot()
    })
  })

  describe('Club', () => {
    it('sets', () => {
      expect(GLOBAL_ACTIONS.CLUB.SET({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('gets', () => {
      expect(GLOBAL_ACTIONS.CLUB.GET({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('succeeds', () => {
      expect(GLOBAL_ACTIONS.CLUB.SUCCESS({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('errors', () => {
      expect(GLOBAL_ACTIONS.CLUB.ERROR({
        error: 'some error here'
      })).toMatchSnapshot()
    })

    it('clears', () => {
      expect(GLOBAL_ACTIONS.CLUB.CLEAR()).toMatchSnapshot()
    })
  })

  describe('Location', () => {
    it('sets', () => {
      expect(GLOBAL_ACTIONS.LOCATIONS.SET_ALL({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('gets', () => {
      expect(GLOBAL_ACTIONS.LOCATIONS.GET({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('succeeds', () => {
      expect(GLOBAL_ACTIONS.LOCATIONS.SUCCESS({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('errors', () => {
      expect(GLOBAL_ACTIONS.LOCATIONS.ERROR({
        error: 'some error here'
      })).toMatchSnapshot()
    })

    it('clears', () => {
      expect(GLOBAL_ACTIONS.LOCATIONS.CLEAR()).toMatchSnapshot()
    })

    it('selects', () => {
      expect(GLOBAL_ACTIONS.LOCATIONS.SELECT({
        data: { clubId: undefined, locationId: 1974 }
      })).toMatchSnapshot()
    })

    it('sets', () => {
      expect(GLOBAL_ACTIONS.LOCATIONS.SET({
        data: { name: 'Valhalla', value: '1' }
      })).toMatchSnapshot()
    })
  })

  describe('Data', () => {
    it('gets', () => {
      expect(GLOBAL_ACTIONS.DATA.GET({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('succeeds', () => {
      expect(GLOBAL_ACTIONS.DATA.SUCCESS({
        data: 'some data here'
      })).toMatchSnapshot()
    })

    it('errors', () => {
      expect(GLOBAL_ACTIONS.DATA.ERROR({
        error: 'some error here'
      })).toMatchSnapshot()
    })

    it('clears', () => {
      expect(GLOBAL_ACTIONS.DATA.CLEAR()).toMatchSnapshot()
    })
  })
})
