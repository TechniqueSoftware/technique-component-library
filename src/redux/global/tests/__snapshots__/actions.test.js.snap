// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GLOBAL ACTIONS Club clears 1`] = `
Object {
  "type": "global/CLUB/CLEAR",
}
`;

exports[`GLOBAL ACTIONS Club errors 1`] = `
Object {
  "payload": Object {
    "error": "some error here",
  },
  "type": "global/CLUB/ERROR",
}
`;

exports[`GLOBAL ACTIONS Club gets 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/CLUB/GET",
}
`;

exports[`GLOBAL ACTIONS Club sets 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/CLUB/SET",
}
`;

exports[`GLOBAL ACTIONS Club succeeds 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/CLUB/SUCCESS",
}
`;

exports[`GLOBAL ACTIONS Data clears 1`] = `
Object {
  "type": "global/DATA/CLEAR",
}
`;

exports[`GLOBAL ACTIONS Data errors 1`] = `
Object {
  "payload": Object {
    "error": "some error here",
  },
  "type": "global/DATA/ERROR",
}
`;

exports[`GLOBAL ACTIONS Data gets 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/DATA/GET",
}
`;

exports[`GLOBAL ACTIONS Data succeeds 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/DATA/SUCCESS",
}
`;

exports[`GLOBAL ACTIONS Loading finishes 1`] = `
Object {
  "type": "global/LOADING/FINISHED",
}
`;

exports[`GLOBAL ACTIONS Location clears 1`] = `
Object {
  "type": "global/LOCATIONS/CLEAR",
}
`;

exports[`GLOBAL ACTIONS Location errors 1`] = `
Object {
  "payload": Object {
    "error": "some error here",
  },
  "type": "global/LOCATIONS/ERROR",
}
`;

exports[`GLOBAL ACTIONS Location gets 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/LOCATIONS/GET",
}
`;

exports[`GLOBAL ACTIONS Location selects 1`] = `
Object {
  "payload": Object {
    "data": Object {
      "clubId": undefined,
      "locationId": 1974,
    },
  },
  "type": "global/LOCATIONS/SELECT",
}
`;

exports[`GLOBAL ACTIONS Location sets 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/LOCATIONS/SET_ALL",
}
`;

exports[`GLOBAL ACTIONS Location sets 2`] = `
Object {
  "payload": Object {
    "data": Object {
      "name": "Valhalla",
      "value": "1",
    },
  },
  "type": "global/LOCATIONS/SET",
}
`;

exports[`GLOBAL ACTIONS Location succeeds 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/LOCATIONS/SUCCESS",
}
`;

exports[`GLOBAL ACTIONS User clears 1`] = `
Object {
  "type": "global/USER/CLEAR",
}
`;

exports[`GLOBAL ACTIONS User errors 1`] = `
Object {
  "payload": Object {
    "error": "some error here",
  },
  "type": "global/USER/ERROR",
}
`;

exports[`GLOBAL ACTIONS User gets 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/USER/GET",
}
`;

exports[`GLOBAL ACTIONS User sets 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/USER/SET_LOGGED_IN",
}
`;

exports[`GLOBAL ACTIONS User succeeds 1`] = `
Object {
  "payload": Object {
    "data": "some data here",
  },
  "type": "global/USER/SUCCESS",
}
`;
